packages/test/stochastic-test-utils/src/random.ts

```diff
@@ -2,24 +2,80 @@
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
  */
-import Random from "random-js";
+
+/* eslint-disable no-bitwise */
+
+import * as distribution from "./distributions";
+import { IRandom } from "./types";
+import { XSadd } from "./xsadd";
+
+// Constructs a compliant UUID version 4 from four 32b integers.  UUID version 4 contains
+// 6 predetermined bits (bits 48..51 for the version and bits 64..65 for the variant).
+// Consequently, only 122 of the provided 128 bits are used.
+export function makeUuid4(u32_0: number, u32_1: number, u32_2: number, u32_3: number) {
+    const hex = (value: number, digits: number) => value.toString(16).padStart(digits, "0");
+
+    return `${
+        hex(u32_0, 8)
+    }-${
+        hex(u32_1 >>> 16, 4)
+    }-${
+        // Discard low 4 bits and insert version '0b100'.
+        hex(((u32_1 << 16) >>> 20) | 0x4000, 4)
+    }-${
+        // Insert variant '0b10', discarding the low 2 bits.
+        hex((u32_2 >>> 18) | 0x8000, 4)
+    }-${
+        // Use the 2 bits that were discarded above and instead discard weak low bits.
+        hex((u32_2 << 14) >>> 16, 4)
+    }${
+        hex(u32_3, 8)
+    }`;
```

Comment:
I chose which low bits to discard based on convenience.  Discarding the weakest bit from each u32 would be better, but not worth it for our application...  but I should probably leave a comment.